src/sparseml/transformers/compression/compressors/sparse_bitmask.py

```diff
@@ -54,46 +61,77 @@ def compress(self, model_state: Dict[str, Tensor]) -> Dict[str, Tensor]:
             f"Compressing model with {len(model_state)} parameterized layers..."
         )
         for name, value in tqdm(model_state.items()):
-            bitmask_tensor = BitmaskTensor(value)
+            bitmask_tensor = BitmaskTensor.from_dense(value)
             compressed_dict |= bitmask_tensor.dict(name_prefix=name)
 
         return compressed_dict
 
-    def decompress(self, model_state: Dict[str, Tensor]) -> Dict[str, Tensor]:
+    def decompress(self, model_path: str) -> Dict[str, Tensor]:
```

Comment:
could we have a variant of `decompress` method that yields weights keys and values? This would be much better and efficient fit for vLLM integration.

Reply:
Yes we could definitely write a version of decompress that returns an iterator. I think thats best left as a followup though since the current implementation does work with vLLM

Reply:
Going back on this actually @dbogunowicz, in order to reduce memory footprint I think it makes sense to just yield in this function. Pushing a change shortly